modules/ximgproc/include/fast_hough_transform.hpp

```diff
@@ -0,0 +1,165 @@
+  FHT_AVE = 3   //< Binary average operation. The constant specifies the binary average operation
+                //< @f$ f @f$ that is defined as follows: @f[ f(x, y) = \frac{x + y}{2} @f]
+};
+
+/**
+* @brief   Specifies to do or not to do skewing of Hough transform image
+* @details The enum specifies to do or not to do skewing of Hough transform image
+* so it would be no cycling in Hough transform image through borders of image.
+*/
+enum HoughDeskewOption
+{
+  HDO_RAW    = 0, //< Use raw cyclic image
+  HDO_DESKEW = 1  //< Prepare deskewed image
+};
+
+/**
+ * @brief   Specifies the degree of rules validation.
+ * @details The enum specifies the degree of rules validation. This can be used,
+ *          for example, to choose a proper way of input arguments validation.
+ */
+typedef enum {
+  RO_STRICT          = 0x00,  ///< Validate each rule in a proper way.
+  RO_IGNORE_BORDERS  = 0x01,  ///< Skip validations of image borders.
+} RulesOption;
+
+/**
+* @brief   Calculates 2D Fast Hough transform of an image.
+* @param   dst         The destination image, result of transformation.
+* @param   src         The source (input) image.
+* @param   dstMatDepth The depth of destination image
+* @param   op          The operation to be applied, see cv::HoughOp
+* @param   angleRange  The part of Hough space to calculate, see cv::AngleRangeOption
+* @param   makeSkew    Specifies to do or not to do image skewing, see cv::HoughDeskewOption
+*
+* The function calculates the fast Hough transform for full, half or quarter
+* range of angles.
+*/
+CV_EXPORTS_W void FastHoughTransform( InputArray src,
+                                      OutputArray dst,
+                                      int         dstMatDepth,
+                                      int         angleRange = ARO_315_135,
+                                      int         op = FHT_ADD,
+                                      int         makeSkew = HDO_DESKEW );
+
+/**
+* @brief   Calculates coordinates of line segment corresponded by point in Hough space.
+* @param   line       Coordinates of line segment corresponded by point in Hough space.
+* @param   houghPoint  Point in Hough space.
+* @param   srcImgInfo The source (input) image of Hough transform.
+* @param   angleRange  The part of Hough space where point is situated, see cv::AngleRangeOption
+* @param   makeSkew    Specifies to do or not to do image skewing, see cv::HoughDeskewOption
+* @param   rules       Specifies strictness of line segment calculating, see cv::RulesOption
+* @remarks If rules parameter set to RO_STRICT
+           then returned line cut along the border of source image.
+* @remarks If rules parameter set to RO_WEAK then in case of point, which belongs
+           the incorrect part of Hough image, returned line will not intersect source image.
+*
+* The function calculates coordinates of line segment corresponded by point in Hough space.
+*/
+CV_EXPORTS_W void HoughPoint2Line( OutputArray  line,
```

Comment:
shall we return Vec4i instead of using heavy OutputArray structure for a single line? Or maybe make it a batch function "HoughPoints2Lines"? Also, the output parameters should be put after input ones (and before the algorithm parameters), so if you decide not to return line but use the output parameter, move it after srcImgInfo. And srcImgInfo should preferably be made InputArray